agent/engine/docker_image_manager_test.go

```diff
@@ -1002,20 +1004,215 @@ func TestNonECSImageAndContainersCleanupRemoveImage(t *testing.T) {
 			ID: "1",
 			State: &types.ContainerState{
 				Status:     "exited",
-				FinishedAt: time.Now().AddDate(0, -2, 0).String(),
+				FinishedAt: time.Now().AddDate(0, -2, 0).Format(time.RFC3339Nano),
+			},
+		},
+	}
+
+	inspectImageResponse := &types.ImageInspect{
+		Size: 4096,
+	}
+
+	listImagesResponse := dockerapi.ListImagesResponse{
+		ImageIDs: []string{"sha256:qwerty1"},
+		RepoTags: []string{"tester"},
+	}
+
+	client.EXPECT().ListContainers(gomock.Any(), gomock.Any(), dockerclient.ListImagesTimeout).Return(listContainersResponse).AnyTimes()
+	client.EXPECT().InspectContainer(gomock.Any(), gomock.Any(), dockerclient.InspectContainerTimeout).Return(inspectContainerResponse, nil).AnyTimes()
+	client.EXPECT().RemoveContainer(gomock.Any(), gomock.Any(), dockerclient.RemoveContainerTimeout).Return(nil).Times(1)
+	client.EXPECT().ListImages(gomock.Any(), dockerclient.ListImagesTimeout).Return(listImagesResponse).AnyTimes()
+	client.EXPECT().InspectImage(listImagesResponse.RepoTags[0]).Return(inspectImageResponse, nil).Times(1)
+	client.EXPECT().RemoveImage(gomock.Any(), listImagesResponse.RepoTags[0], dockerclient.RemoveImageTimeout).Return(nil).Times(1)
+
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	imageManager.removeUnusedImages(ctx)
+
+	if len(listContainersResponse.DockerIDs) != 1 {
+		t.Error("Error removing containers ids")
```

Comment:
you can use `assert.Len` https://godoc.org/github.com/stretchr/testify/assert#Assertions.Len / `assert.Equal` for this and below ones https://godoc.org/github.com/stretchr/testify/assert#Equal

Reply:
👍